gson/src/test/java/com/google/gson/internal/bind/DefaultDateTypeAdapterTest.java

```diff
@@ -223,6 +225,36 @@ public void testUnexpectedToken() throws Exception {
     }
   }
 
+  @Test
+  public void testGsonDateFormat() {
+    TimeZone originalTimeZone = TimeZone.getDefault();
+    // Set the default timezone to UTC
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+    try {
+      Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd HH:mm z").create();
+      Date originalDate = new Date(0);
+
+      // Serialize the date object
+      String json = gson.toJson(originalDate);
+      assertEquals("\"1970-01-01 00:00 UTC\"", json);
```

Comment:
Please use Truth for assertions here:
```suggestion
      assertThat(json).isEqualTo("\"1970-01-01 00:00 UTC\"");
```